DOCS: CE-661 Refactor CNI plugin content (#23707)
hashicorp/nomad: website/content/docs/networking/index.mdx

}
```

Nomad reserves a random port in the client between [`min_dynamic_port`](/nomad/docs/configuration/client#min_dynamic_port) and [`max_dynamic_port`](/nomad/docs/configuration/client#max_dynamic_port) that has not been allocated yet. Nomad then creates a port mapping from the host network interface to the allocation.

[comment-image-source]:  https://drive.google.com/file/d/1q4a2ab0TyLEPdWiO2DIianAPWuPqLqZ4/view?usp=share_link

[![Nomad Port Mapping](/img/networking/port_mapping.png)](/img/networking/port_mapping.png)

Tasks can access the selected port number using the [`NOMAD_PORT_<label>`](/nomad/docs/runtime/environment#network-related-variables) environment variable to bind and expose the workload at the client's IP address and the given port.

> The specific configuration process depends on what you are running. However, usually you use a [`template`](/nomad/docs/job-specification/template#template-examples) to create a configuration file such as the follwowing:

Reviewer: ```suggestion
The specific configuration process depends on what you are running. However, usually you use a [`template`](/nomad/docs/job-specification/template#template-examples) to create a configuration file or pass it directly via command line arguments such as the following:
```

}
```

> You you can pass configuration via command line arguments.

Reviewer: ```suggestion
```

The tasks in the rightmost allocation are not able to communicate with each other using their loopback interface because they were placed in different network namespaces.

Since the group `network.mode` is `bridge`, Nomad creates the pause container to establish a shared network namespace for all tasks, but setting the task-level `network_mode` to `bridge` places the task in a different namespace. This prevents, for example, a task from communicating with its sidecar proxy in a service mesh deployment.

Refer to the [`network_mode`](/nomad/docs/drivers/docker#network_mode) documentation and the [Networking](/nomad/docs/drivers/docker#networking) section for more information.

-> **Note:** Docker Desktop in non-Linux environments runs a local virtual machine, adding an extra layer of indirection. Refer to the
   [FAQ](/nomad/docs/faq#q-how-to-connect-to-my-host-network-when-using-docker-desktop-windows-and-macos) for more details.

## Container Network Interface (CNI) plugins

Nomad has built-in support for scheduling compute resources such as CPU, memory, and networking. Nomad's network plugin support extends

> this to allow scheduling tasks with specialty network configurations. Network plugins are third-party plugins that conform to the [Container Network Interface (CNI) [specification](https://www.cni.dev/docs/spec/).

Reviewer 2: Stray bracket:

```suggestion
this to allow scheduling tasks with specialty network configurations. Network plugins are third-party plugins that conform to the Container Network Interface (CNI) [specification](https://www.cni.dev/docs/spec/).
```

> this to allow scheduling tasks with specialty network configurations. Network plugins are third-party plugins that conform to the [Container Network Interface (CNI) [specification](https://www.cni.dev/docs/spec/).

Reviewer: ```suggestion
this to allow scheduling tasks with specialty network configurations. Network plugins are third-party plugins that conform to the [Container Network Interface (CNI) specification](https://www.cni.dev/docs/spec/).
```

To access a service from another container you can reference the service name
directly, for example using `postgres://db:5432`. In order to enable this

> pattern, Docker Compose includes an [internal DNS services](https://docs.docker.com/config/containers/container-networking/#dns-services) and a load balancer that is transparent to user. When running in Swarm mode, Docker Compose also requires an overlay network to route requests across hosts.

Reviewer: ```suggestion
pattern, Docker Compose includes an [internal DNS services](https://docs.docker.com/config/containers/container-networking/#dns-services) and a load balancer that is transparent to the user. When running in Swarm mode, Docker Compose also requires an overlay network to route requests across hosts.
```

      targetPort: 9376
```

To access the Service you use a FQDN such as `my-service.prod.svc.cluster.local`. This name is resolved by the [DNS service](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/) which is an add-on that runs in all nodes. Along with this service, each node also runs a [`kube-proxy`](https://kubernetes.io/docs/concepts/overview/components/#kube-proxy) instance to distribute requests to all Pods matched by the Service.

You can use the same FQDN networking style with Nomad using [Consul's DNS
interface](/consul/docs/services/discovery/dns-overview) and configuring your clients with [DNS
forwarding](/consul/tutorials/networking/dns-forwarding), and deploying a [load
balancer](/nomad/tutorials/load-balancing).

Another key difference from Nomad is that in Kubernetes and Docker Compose each
container has its own IP address, requiring a virtual network to map physical

> IP addresses to virtual ones. In case of Docker Compose in Swarm mode an

Reviewer: ```suggestion
IP addresses to virtual ones. In the case of Docker Compose in Swarm mode, an
```

[`overlay`](https://docs.docker.com/network/overlay/) is also required to enable traffic across multiple hosts. This allows multiple containers running the same service to listen on
the same port number.

> In Nomad, allocations use the IP address of the client they are running and are
> assigned random port numbers and so Nomad service discovery with DNS uses

Reviewer: ```suggestion
In Nomad, allocations use the IP address of the client in which they are running and are
assigned random port numbers. Nomad service discovery with DNS uses
```
